test/server/controller.spec.js

```diff
@@ -1100,101 +1441,126 @@ describe('redirection when running behind a proxy', () => {
+          'Bypassing proxy server is not allowed');
+        done();
+      });
+      if (obj.method === "POST") {
+        req.write(JSON.stringify({"format":"bam"}), () => {
+          req.end();
+        });
+      } else {
+        req.end();
+      }
     });
-    req.end();
-  });
 
-  it('unknown proxy is not allowed', (done) => {
-    let options = {
-      socketPath: socket,
-      path:       serverPath,
-      headers:    {'X-Forwarded-Host': 'myserver.com:9090'},
-      method:    'GET'};
-    let req = http.request(options);
-    req.on('response', (res) => {
-      expect(res.statusCode).toEqual(403);
-      expect(res.statusMessage).toEqual(
-        'Unknown proxy http://myserver.com:9090');
-      done();
+    it(`unknown proxy is not allowed for ${optionsList.method}`, (done) => {
+      let obj = JSON.parse(JSON.stringify(optionsList));
+      obj.path = serverPath;
+      obj.headers =  {'X-Forwarded-Host': 'myserver.com:9090'};
+      let req = http.request( obj );
+      req.on('response', (res) => {
+        expect(res.statusCode).toEqual(403);
+        expect(res.statusMessage).toEqual(
+          'Unknown proxy http://myserver.com:9090');
+        done();
+      });
+      if (obj.method === "POST") {
+        req.write(JSON.stringify({"format":"bam"}), () => {
+          req.end();
+        });
+      } else {
+        req.end();
+      }
     });
-    req.end();
-  });
 
-  it('Redirection to one of known proxies', (done) => {
-    let url = `http://myserver.com:3456/path3/sample?accession=${id}&format=BAM`;
-    let options = {
-      socketPath: socket,
-      path:       serverPath,
-      headers:    {'X-Forwarded-Host': 'myserver.com:3456'},
-      method:    'GET'};
-    let req = http.request(options);
-    req.on('response', (res) => {
-      var body = '';
-      expect(res.statusCode).toEqual(200);
-      res.on('data', (d) => { body += d;});
-      res.on('end', () => {
-        expect(JSON.parse(body)).toEqual({
-          htsget: {
-            format: 'BAM',
-            urls: [{'url': url}]
-          }
+    it(`Redirection to one of known proxies for ${optionsList.method}`, (done) => {
+      let obj = JSON.parse(JSON.stringify(optionsList));
+      obj.path = serverPath;
+      obj.headers =  {'X-Forwarded-Host': 'myserver.com:3456'};
+      let url = `http://myserver.com:3456/path3/sample?accession=${id}&format=BAM`;
+      let req = http.request( obj );
+      req.on('response', (res) => {
+        var body = '';
+        expect(res.statusCode).toEqual(200);
+        res.on('data', (d) => { body += d;});
+        res.on('end', () => {
+          expect(JSON.parse(body)).toEqual({
+            htsget: {
+              format: 'BAM',
+              urls: [{'url': url}]
+            }
+          });
+          done();
         });
-        done();
       });
+      if (obj.method === "POST") {
+        req.write(JSON.stringify({"format":"bam", "accession":id}), () => {
```

Comment:
I was not expecting you to need to specify a `accession` here. Can you delete it and see if test still runs?